app/src/pages/playground/PlaygroundOutput.tsx

```
@@ -265,7 +268,26 @@ function PlaygroundOutputText(props: PlaygroundInstanceProps) {
         }
       : {};
 
-  const invocationParameters: InvocationParameters = {};
+  const invocationParametersSchema = getInvocationParametersSchema({
+    modelProvider: instance.model.provider,
+    modelName: instance.model.modelName || "default",
+  });
+
+  // Constrain the invocation parameters to the schema.
+  // This prevents us from sending invalid parameters to the LLM.
+  let invocationParameters: InvocationParameters = {
+    ...instance.model.invocationParameters,
+  };
+
+  const valid = invocationParametersSchema.safeParse(invocationParameters);
+  if (!valid.success) {
+    // TODO(apowell): We should fail open here and display all inputs.
+    // eslint-disable-next-line no-console
+    console.error(valid.error);
+    // Fall back to the model's invocation parameters.
+    invocationParameters = instance.model.invocationParameters;
+  }
```

okay so this is happening before we send the parameters down to the server, i wonder if we should do this here or just let it fail. Is the thought that If we constrain the form properly seems like this should never be hit correct?

---

Yeah I am thinking we let this fail now. I wrote this before I did all the magic parsing/ form generation inside of the form itself

---

Updated the comments to more clearly explain whats going on